### PR TITLE
Modelcard++ OT (ICS) & DGA

### LINKED ISSUE #50: [DOC]: Update Model Cards to Model Card ++ format
### How would you describe the priority of this documentation request

High

### Please provide a link or source to the relevant docs

https://github.com/nv-morpheus/morpheus-experimental/blob/branch-23.07/anomalous-auth-detection/README.md

### Describe the problems in the documentation

Model cards need to specifically address bias, explainability, privacy, safety and security per Nvidia's model card ++ format.

Model creators should update their own model cards.

- [ ] anomalous-auth-detection @tzemicheal 
- [ ] appshield-dga-detection @raykallen 
- [ ] asset-clustering @avinashvem
- [ ] dga-detection @gbatmaz 
- [ ] ids-detection @tzemicheal 
- [ ] log-sequence-ad @tzemicheal 
- [ ] operational-technology @gbatmaz 
- [ ] phishing-url-detection @gbatmaz 
- [ ] string-resemblance-grouping @shawn-davis 

### (Optional) Propose a correction

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I have searched the [open documentation issues](https://github.com/nv-morpheus/morpheus-experimental/issues?q=is%3Aopen+is%3Aissue+label%3Adoc) and have found no duplicates for this bug report

### PR DESCRIPTION
**Pull request by gbatmaz**
## Description

Appshield URL model card issue will be discussed offline.
## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- [ ] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by hsin-c** [comment]
Let's make a top-level model-cards directory and move all the model card information into their own files (one new file per model like what we have in Morpheus proper)

File: dga-detection/README.md [hunk 1]
```diff
@@ -1,53 +1,245 @@
...
-pip install -r requirements.txt
-```
+## References(s):
+* Cho et al. 2014 https://arxiv.org/abs/1406.1078 <br> 
 
-### Training
+## Model Architecture: 
+**Architecture Type:** 
+* Recurrent Neural Network <br>
+**Network Architecture:**
```
Comment (hsin-c) [inline dga-detection/README.md:20]: The same formatting comment from Morpheus proper applies here. All questions should be in a new line. 
(It's now concatenated with the previous line: 
image
)

File: dga-detection/README.md [hunk 2]
```diff
@@ -1,53 +1,245 @@
...
 
-#### Training batch size
-10000
+**Supported Operating System(s):** <br>
+* Linux <br>
 
-#### GPU model
-V100
+## Model Version(s): 
+* [v1]  <br>
```
Comment (hsin-c) [inline dga-detection/README.md:50]: We can remove the `[]`

File: dga-detection/README.md [hunk 3]
```diff
@@ -1,53 +1,245 @@
...
 
-To train the model run the following script under working directory.
-```bash
-cd ${MORPHEUS_EXPERIMENTAL_ROOT}/dga-detection/training-tuning-inference
+**Link:** 
+* https://github.com/nv-morpheus/morpheus-experimental/blob/branch-23.07/dga-detection/datasets/dga-training-data.csv  <br>
+**Properties (Quantity, Dataset Descriptions, Sensor(s)):**
+* Domain names <br>
+**Dataset License:**
+* N/A <br>
```
Comment (hsin-c) [inline dga-detection/README.md:61]:
```
**Dataset License:**  
* [Apache 2.0](http://www.apache.org/licenses/LICENSE-2.0) <br>
```

File: dga-detection/README.md [hunk 4]
```diff
@@ -1,53 +1,245 @@
...
+
+* Not Applicable
+
+### What is the age balance of the model validation data?
+
+* Not Applicable
+
+### What is the language balance of the model validation data?
+
+* Not Applicable
```
Comment (hsin-c) [inline dga-detection/README.md:95]: Let's say `English: 100%` if the dataset is in English (the domains probably are?)

File: dga-detection/README.md [hunk 5]
```diff
@@ -1,53 +1,245 @@
...
+* N/A
+
+### Is data in dataset traceable?
+* N/A
+
+### Scanned for malware?
+* N/A
+
+### Are we able to identify and trace source of dataset?
+* N/A
```
Comment (hsin-c) [inline dga-detection/README.md:239]: If we made and own the dataset, we can say `Yes` here.

File: dga-detection/README.md [hunk 6]
```diff
@@ -1,53 +1,245 @@
...
+
+### If PII collected for the development of this AI model, was it minimized to only what was required? 
+* N/A
+
+### Is data in dataset traceable?
+* N/A
+
+### Scanned for malware?
+* N/A
```
Comment (hsin-c) [inline dga-detection/README.md:236]: As it's probably not scanned-- let's say `* No`

File: dga-detection/README.md [hunk 7]
```diff
@@ -1,53 +1,245 @@
...
+* Neither
+
+### Was consent obtained for any PII used?
+* N/A
+
+### Protected classes used to create this model? (The following were used in model the model's training:)
+* N/A
+
+### How often is dataset reviewed?
+* N/A
```
Comment (hsin-c) [inline dga-detection/README.md:223]: It's reviewed when we made the PR to add it in. Let's say 
`
The dataset is initially reviewed upon addition, and subsequent reviews are conducted as needed or upon request for any changes.
`

File: operational-technology/README.md [hunk 8]
```diff
@@ -1,94 +1,249 @@
...
+## References(s):
+* https://sites.google.com/a/uah.edu/tommy-morris-uah/ics-data-sets <br>
+* http://www.ece.uah.edu/~thm0009/icsdatasets/PowerSystem_Dataset_README.pdf <br>
 
-### Model Architecture
-XGBoost Classifier
+## Model Architecture: 
+**Architecture Type:** 
+* Gradient Boosting <br>
+**Network Architecture:**
```
Comment (hsin-c) [inline operational-technology/README.md:15]: The same formatting comment from Morpheus proper applies here. All questions should be in a new line.

File: operational-technology/README.md [hunk 9]
```diff
@@ -1,94 +1,249 @@
...
-#### Model accuracy
+**Supported Hardware Platform(s):** <br>
+* Ampere/Turing <br>
 
-The label distribution in the dataset is imbalanced, so we do not use the accuracy score. Instead, we use F1 weighted as the metric. The F1 score was over 0.91 on a test set.
+**Supported Operating System(s):** <br>
+* Linux <br>
 
+## Model Version(s): 
+* [v1]  <br>
```
Comment (hsin-c) [inline operational-technology/README.md:54]: We can remove the `[]`

File: operational-technology/README.md [hunk 10]
```diff
@@ -1,94 +1,249 @@
...
-python ot-xgboost-train.py \
-    --model ../models/ot-xgboost-20230207.pkl
-```
-This will download the data (if it is not present) and train a model with a training set, and it will save a model under the `models` directory.
+**Link:** 
+* http://www.ece.uah.edu/~thm0009/icsdatasets/triple.7z <br>
+**Properties (Quantity, Dataset Descriptions, Sensor(s)):**
+* There are 78377 rows in the dataset. <br>
+**Dataset License:**
+* N/A <br>
```
Comment (hsin-c) [inline operational-technology/README.md:65]: Please add the data license in if we can (the datasets that we’re not hosting should have the license of where they’re being pulled from)

File: operational-technology/README.md [hunk 11]
```diff
@@ -1,94 +1,249 @@
...
 
-### Inference
+## Evaluation Dataset:
+**Link:** 
+* http://www.ece.uah.edu/~thm0009/icsdatasets/triple.7z <br>
+**Properties (Quantity, Dataset Descriptions, Sensor(s)):**
+* There are 78377 rows in the dataset. <br>
+**Dataset License:**
+* N/A <br>
```
Comment (hsin-c) [inline operational-technology/README.md:73]: Please add the data license in if we can (the datasets that we’re not hosting should have the license of where they’re being pulled from)

File: operational-technology/README.md [hunk 12]
```diff
@@ -1,94 +1,249 @@
...
+* N/A
+
+### Is data in dataset traceable?
+* N/A
+
+### Scanned for malware?
+* N/A
+
+### Are we able to identify and trace source of dataset?
+* N/A
```
Comment (hsin-c) [inline operational-technology/README.md:243]: Let's say `Yes` since we do know where this dataset is being hosted

File: operational-technology/README.md [hunk 13]
```diff
@@ -1,94 +1,249 @@
...
+* Neither
+
+### Was consent obtained for any PII used?
+* N/A
+
+### Protected classes used to create this model? (The following were used in model the model's training:)
+* N/A
+
+### How often is dataset reviewed?
+* N/A
```
Comment (hsin-c) [inline operational-technology/README.md:227]: Do we know how often this public dataset is being reviewed?

File: operational-technology/README.md [hunk 14]
```diff
@@ -1,94 +1,249 @@
...
+* No
+
+### Is the model and dataset compliant with National Classification Management Society (NCMS)?
+* No
+
+### Are there explicit model and dataset restrictions?
+* No
+
+### Are there access restrictions to systems, model, and data?
+* N/A
```
Comment (hsin-c) [inline operational-technology/README.md:208]: I think we can say `* No` here

**Review by hsin-c — CHANGES_REQUESTED** [review]

File: dga-detection/README.md (hunk 4 above)
Comment (gbatmaz) [reply dga-detection/README.md]: I've added a comment to indicate domains can be any language

File: dga-detection/README.md (hunk 1 above)
Comment (gbatmaz) [reply dga-detection/README.md]: done

File: dga-detection/README.md (hunk 5 above)
Comment (gbatmaz) [reply dga-detection/README.md]: done

File: dga-detection/README.md (hunk 7 above)
Comment (gbatmaz) [reply dga-detection/README.md]: Added that statement

File: operational-technology/README.md (hunk 10 above)
Comment (gbatmaz) [reply operational-technology/README.md]:
```Please cite the following paper when using these datasets.

Morris, T., Thornton, Z., Turnipseed, I.,  Industrial Control System Simulation and Data Logging for Intrusion Detection System Research. 7th Annual Southeastern Cyber Security Summit. Huntsvile, AL. June 3 - 4, 2015. [pdf](http://www.google.com/url?q=http%3A%2F%2Fwww.ece.uah.edu%2F~thm0009%2Ficsdatasets%2Fcyberhuntsvillepaper_v4.pdf&sa=D&sntz=1&usg=AOvVaw0Tnq7QQfYiD7GO-mqAAt81)```
I thinks open for any use. They say this without mentioning any explicit license

Comment (gbatmaz) [reply operational-technology/README.md]: hence I put the source URL

File: operational-technology/README.md (hunk 11 above)
Comment (gbatmaz) [reply operational-technology/README.md]: I thinks open for any use. They say this without mentioning any explicit license
"Please cite the following paper when using these datasets.

Morris, T., Thornton, Z., Turnipseed, I., Industrial Control System Simulation and Data Logging for Intrusion Detection System Research. 7th Annual Southeastern Cyber Security Summit. Huntsvile, AL. June 3 - 4, 2015. [pdf]
" Hence I put the source UrL

File: operational-technology/README.md (hunk 14 above)
Comment (gbatmaz) [reply operational-technology/README.md]: added

File: operational-technology/README.md (hunk 13 above)
Comment (gbatmaz) [reply operational-technology/README.md]: I put unknown

File: operational-technology/README.md (hunk 12 above)
Comment (gbatmaz) [reply operational-technology/README.md]: yes :)

File: operational-technology/README.md (hunk 8 above)
Comment (gbatmaz) [reply operational-technology/README.md:24]: fixed format

File: dga-detection/README.md (hunk 6 above)
Comment (gbatmaz) [reply dga-detection/README.md:297]: done

**Review by hsin-c — APPROVED** [review]
